src/App.jsx

```diff
@@ -7,15 +9,28 @@ import { Navbar } from './components/Navbar'
 import { Home } from './pages/Home'
 import { Player } from './pages/Player'
 
-
 function App() {
 
+  const containerRef = useRef()
+  const [containerWidth, setContainerWidth] = useState(0)
+
```

Comment:
I think you're over thinking this. With MUI you can access the viewports for any CSS. So if you want smaller padding on a container you can do sx={{ margin: { xs: 1, md: 2}}}